crates/holochain/src/conductor/conductor.rs

```diff
@@ -1499,29 +1513,43 @@ mod app_impls {
                 self.clone().register_dna(dna).await?;
             }
 
-            let cell_ids: Vec<_> = cells_to_create
-                .iter()
-                .map(|(cell_id, _)| cell_id.clone())
-                .collect();
+            if do_genesis {
+                let cell_ids: Vec<_> = cells_to_create
+                    .iter()
+                    .map(|(cell_id, _)| cell_id.clone())
+                    .collect();
 
-            let genesis_result =
-                crate::conductor::conductor::genesis_cells(self.clone(), cells_to_create).await;
+                let genesis_result =
+                    crate::conductor::conductor::genesis_cells(self.clone(), cells_to_create).await;
 
-            if genesis_result.is_ok() || ignore_genesis_failure {
-                let roles = ops.role_assignments;
-                let app = InstalledAppCommon::new(installed_app_id, agent_key, roles, manifest)?;
+                if genesis_result.is_ok() || ignore_genesis_failure {
+                    let roles = ops.role_assignments;
+                    let app =
+                        InstalledAppCommon::new(installed_app_id, agent_key, roles, manifest)?;
 
-                // Update the db
-                let stopped_app = self.add_disabled_app_to_db(app).await?;
+                    // Update the db
+                    let stopped_app = self.add_disabled_app_to_db(app).await?;
 
-                // Return the result, which be may an error if no_rollback was specified
-                genesis_result.map(|()| stopped_app)
-            } else if let Err(err) = genesis_result {
-                // Rollback created cells on error
-                self.remove_cells(&cell_ids).await;
-                Err(err)
+                    // Return the result, which be may an error if no_rollback was specified
+                    genesis_result.map(|()| stopped_app.into())
+                } else if let Err(err) = genesis_result {
+                    // Rollback created cells on error
+                    self.remove_cells(&cell_ids).await;
+                    Err(err)
+                } else {
+                    unreachable!()
```

Comment:
I agree, but I didn't do too much restructuring on this one because the DPKI branch heavily refactored this function already, so I'm going to have to rewrite this when that finally gets merged in.

Reply:
That seems like an undesirable situation to be in. I would really like to see the other PR get merged then so that we can review new PRs without having to worry about multiple versions of the code. I've spent some time this morning on the flaky test PR but every time we get past one there's another. We've let that build up a bit I guess. But I'd like to see if we can get that PR sorted, and get DPKI in ASAP.

Maybe it shouldn't block this but I would really like to be able to review this in a 'clean' state.

Reply:
Yeah, I've been wanting to get DPKI merged for a long time, which is another story.

There is some urgency around this item because there are downstream dependencies in two other teams, which has impacts beyond our codebase, whereas the main benefit of merging the DPKI branch is mostly to help with merge conflicts like this. Jost is already working off of my unmerged DPKI branch, so the fact that it's unmerged is not blocking any work.

So yes I fully agree with you about this being undesirable, and unfortunately I think the urgency of this one is justified, and that the extra work of writing and reviewing the same function twice is justified, undesirable as it may be.